src/sentry/models/statistical_detectors.py

```diff
@@ -0,0 +1,72 @@
+from enum import Enum
+from typing import Sequence, Tuple
+
+from django.db import models
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import (
+    BoundedBigIntegerField,
+    BoundedIntegerField,
+    Model,
+    region_silo_only_model,
+    sane_repr,
+)
+
+
+class RegressionType(Enum):
+    ENDPOINT = 0
+    FUNCTION = 1
+
+    @classmethod
+    def as_choices(cls) -> Sequence[Tuple[int, str]]:
+        return (
+            (cls.ENDPOINT.value, "endpoint"),
+            (cls.FUNCTION.value, "function"),
+        )
+
+
+@region_silo_only_model
+class RegressionGroup(Model):
+    __relocation_scope__ = RelocationScope.Excluded
+
+    # Meta data about the regression group
+    date_added = models.DateTimeField(auto_now_add=True)
+    date_updated = models.DateTimeField(auto_now=True)
+
+    # When the regression started, this is the breakpoint.
+    date_regressed = models.DateTimeField()
+
+    # When the regression resolved.
+    date_resolved = models.DateTimeField(null=True)
+
+    # The version associated with the group. Each time a regression
+    # is detected for the group, we increment the version.
+    version = models.IntegerField()
+
+    # Indiciates if the regression group is active or not. This should
+    # be checked in conjunction with the issue group status to determine
+    # the status of the group as manual status changes do not
+    # propagate from the issue group to here.
+    active = models.BooleanField(default=True)
+
+    project_id = BoundedBigIntegerField(db_index=True)
+
+    type = BoundedIntegerField(choices=RegressionType.as_choices())
+
+    # The fingerprint sent to the issue platform which
+    # accepts a list of strings. This corresponds to the
+    # first string which is a 8-16 char hex value.
+    # Here, it's being stored as it's integer equivalent.
```

Comment:
Why as an integer?

Reply:
I don't have a strong preference here to store it as an integer or string. Just thought it'd be more efficient to store it as an integer as I'll be filtering on this column quite a bit.

If storing it as a string is preferred, happy to switch it.

Reply:
Strings result in fewer translations needed and unless this is going to be 10s of billions of rows the storage delta isn't likely to matter much.